CMakeLists.txt

```diff
@@ -83,6 +82,7 @@ dependent_option(ENABLE_INTERNAL_FLATBUFFERS "Enable internal flatbuffers?")
 dependent_option(ENABLE_INTERNAL_FMT "Enable internal fmt?")
 dependent_option(ENABLE_INTERNAL_NFS "Enable internal libnfs?")
 dependent_option(ENABLE_INTERNAL_PCRE "Enable internal pcre?")
+dependent_option(ENABLE_INTERNAL_RapidJSON "Enable internal rapidjson?")
```

Comment:
What does `dependent_option` do? I was confused if I should use this for my kademlia work.

Reply:
currently as it stands, yes, use it for most all cases.

Its based on the idea of cmakes cmake_dependent_option, however when i started to use cmakes builtin it turns out if you default an option you cant change that (going off my shady memory), so a user couldnt set the define option and alter it.

So i just made a function with the same idea, but it fits our needs a bit better with auto detecting certain kodi specific things

https://github.com/xbmc/xbmc/blob/master/cmake/scripts/common/DependencyOptions.cmake